src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/PackageRestoreDataSource.cs

```diff
@@ -134,6 +135,10 @@ private async Task<bool> RestoreCoreAsync(PackageRestoreUnconfiguredInput value)
                 }
 
                 _latestHash = hash;
+                if (_cycleDetector.ComputeCycleDetection(hash))
+                {
+                    return true;
```

Comment:
We should surface the fact that a loop is suspected to the user, probably via a gold bar.

Something like "An infinite NuGet restore loop is suspected in project _MyProject_. This can happen when a package changes the project's imports, which requires a further restore. The most recent restore was skipped in order to break the cycle."

This serves two purposes:

1. Highlights the issue to the user in case they wish to fix it. It would be great if we could identify the package causing the issue to further help the user.
2. Shows the user that a restore was cancelled, which may be important if they do not actually have a cycle. Perhaps they were just switching between two branches several times, for example.

Reply:
> Shows the user that a restore was cancelled, which may be important if they do not actually have a cycle. Perhaps they were just switching between two branches several times, for example.

Perhaps we add a button to the infobar to allow forcing a restore. Similar to how CPS's project fault infobar has a button to reload the project.

Reply:
InfoBar added.